pkg/pillar/cmd/zedmanager/handledomainmgr.go

```diff
@@ -67,6 +67,30 @@ func MaybeAddDomainConfig(ctx *zedmanagerContext,
 		disk.DisplayName = vrs.DisplayName
 		dc.DiskConfigList = append(dc.DiskConfigList, disk)
 	}
+	// let's fill some of the default values (arguably we may want controller
+	// to do this for us and give us complete config, but it is easier to
+	// fudge DomainConfig for now on our side)
+	if dc.IsContainer() {
+		if dc.Kernel == "" {
+			dc.Kernel = "/hostfs/boot/kernel"
+		}
+		if dc.Ramdisk == "" {
+			dc.Ramdisk = "/usr/lib/xen/boot/runx-initrd"
+		}
+		if dc.ExtraArgs == "" {
+			dc.ExtraArgs = "console=hvc0 root=9p-xen dhcp=1"
+		}
+		if dc.BootLoader == "" {
+			if runtime.GOARCH == "amd64" {
+				dc.BootLoader = "/usr/lib/xen/boot/seabios.bin"
+			} else {
+				dc.BootLoader = "/usr/lib/xen/boot/ovmf.bin"
+			}
+		}
+	}
+	if dc.BootLoader == "" && dc.VirtualizationModeOrDefault() == types.FML {
+		dc.BootLoader = "/usr/lib/xen/boot/ovmf.bin"
```

Comment:
Nit but should this have the amd64 -> seabios code as above?

Reply:
Not sure what you mean -- by this point if we are dealing with a container dc.BootLoader will get filled one way or another by the code above. For a VM FML only applies to x86 and it requires ovmf.bin. The default on x86 is seabios (which is what it is going to be if dc.BootLoader).

Or are you saying that we must always explicitly fill dc.BootLoader out so that we don't rely on qemu defaults?

Reply:
I didn't realize FML is only for x86.